docs/src/prototypes/alerts/AnimatedBannerAlert.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react'
+import { Alert, Animation, SiteVariablesInput, AlertProps } from '@stardust-ui/react'
+
+interface BannerAlertProps extends AlertProps {
+  oof?: boolean
+  critical?: boolean
+  urgent?: boolean
+}
+
+/**
+ * Customized Alert that can express different intents through boolean props:
+ * oof, urgent, critical
+ */
+const BannerAlert: React.FunctionComponent<BannerAlertProps> = props => {
+  const { oof, critical, urgent, ...rest } = props
+  return (
+    <Alert
+      variables={(siteVars: SiteVariablesInput) => {
+        const appWhite = siteVars.colors.grey[50]
+
+        if (oof) {
+          return {
+            color: siteVars.orchid,
+            backgroundColor: '#F9F5F8',
+            borderColor: '#DCADC7',
+          }
+        }
+
+        if (urgent) {
+          return {
+            color: appWhite,
+            backgroundColor: siteVars.red,
+            borderColor: siteVars.red,
+          }
+        }
+
+        if (critical) {
+          return {
+            color: appWhite,
+            backgroundColor: siteVars.gray03,
+            borderColor: siteVars.gray02,
+          }
+        }
+
+        return {}
+      }}
+      {...rest}
+    />
+  )
+}
```

Comment:
It looks confusing that we have a dedicated component to handle only variables.

My proposal let's replace it with a separate function:
```js
const getBannerAlertVariables = (props: BannerAlertProps) => (siteVars: SiteVariablesInput) => {
  const appWhite = siteVars.colors.grey[50]

  if (props.oof) {}
  // ...

  return {}
}
```

And then we can use it:
```js
const { closable, oof, info, urgent, ...rest } = this.props
const { open } = this.state
const variables = getBannerAlertVariables({ oof, info, urgent })

return closable ? (
  <Animation name={open ? '' : 'slideDown'}>
    <Alert {...rest} action={{ icon: 'close', onClick: this.handleClick }} variables={variables} />
   </Animation>
) : (
  <Alert {...rest} variables={variables} />
)
```

Reply:
> It looks confusing that we have a dedicated component to handle only variables.

The point of that is to show users how they can compose the `Alert` to create a `Banner` alert that just introduces new intent options (oof, critical, urgent). Decoupling it like that means they can just copy paste it t without needing the `Animation` part.

What do you think?

Any way to make it more obvious that this is my intention?

Reply:
I don't see there a big difference because there are no complicated logic. Opened to other opinions, @mnajdova @kuzhelov

Reply:
Actually no strong opinion on this one, as it's pretty easy to understand what is happening in both scenarios.